src/core/server/saved_objects/migrationsv2/model/extract_errors.test.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { extractUnknownDocFailureReason } from './extract_errors';
+
+describe('extractUnknownDocFailureReason', () => {
+  it('generates the correct error message', () => {
+    expect(
+      extractUnknownDocFailureReason(
+        [
+          {
+            id: 'unknownType:12',
+            type: 'unknownType',
+          },
+          {
+            id: 'anotherUnknownType:42',
+            type: 'anotherUnknownType',
+          },
+        ],
+        '.kibana_15'
+      )
+    ).toMatchInlineSnapshot(`
+      "Migration failed because documents from unknown types were found. To proceed with the migration, please delete these documents from the \\".kibana_15\\" index.
+      The unknown documents were:
+      - \\"unknownType:12\\" (type: \\"unknownType\\")
+      - \\"anotherUnknownType:42\\" (type: \\"anotherUnknownType\\")
+      You can delete them using the following command:
+      curl -X POST \\"{elasticsearch}/.kibana_15/_bulk?pretty\\" -H 'Content-Type: application/json' -d'
+      { \\"delete\\" : { \\"_id\\" : \\"unknownType:12\\" } }
+      { \\"delete\\" : { \\"_id\\" : \\"anotherUnknownType:42\\" } }
+      '"
+    `);
```

Comment:
Added `curl` command to remove the documents as requested in https://github.com/elastic/kibana/issues/101351#issuecomment-858644229

Still not sure this is really helping as
1. we don't have the ES connection string in the migrator, so end user still have to manually replace it before executing the command
2. if security is enabled, user will also still have to add the proper auth headers for the query to succeed. 
3. Do they have `curl` on windows :trollface: ? 

Wondering if we should really keep this @joshdover?

Reply:
Personally, I think solving (1) may be worth the effort for clarity. I realize it would mean passing the config down several layers, but how feasible is this? Can be done as a follow up as well.

We should make this message a little more cautious though and recommend backing them up first (either with a snapshot or a curl bulk get)